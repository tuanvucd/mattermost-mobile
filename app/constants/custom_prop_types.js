import PropTypes from 'prop-types';

export const Children = PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf([PropTypes.node])]);

export const Style = PropTypes.oneOfType([
    PropTypes.object, // inline style
    PropTypes.number, // style sheet entry
    PropTypes.array,
]);

export default {
    Children,
    Style,
};
