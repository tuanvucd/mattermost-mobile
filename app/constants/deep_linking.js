// Copyright (c) 2015-present GChat, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    CHANNEL: 'channel',
    PERMALINK: 'permalink',
    OTHER: 'other',
};
