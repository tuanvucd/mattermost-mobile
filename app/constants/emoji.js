// Copyright (c) 2015-present GChat, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const ALL_EMOJIS = 'all_emojis';
