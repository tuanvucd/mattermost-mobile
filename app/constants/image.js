

export const IMAGE_MAX_HEIGHT = 350;
export const IMAGE_MIN_DIMENSION = 50;
