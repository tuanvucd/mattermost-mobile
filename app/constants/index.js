

import DeepLinkTypes from './deep_linking';
import DeviceTypes from './device';
import ListTypes from './list';
import NavigationTypes from './navigation';
import PermissionTypes from './permissions';
import ViewTypes, {UpgradeTypes} from './view';

export {
    DeepLinkTypes,
    DeviceTypes,
    ListTypes,
    NavigationTypes,
    PermissionTypes,
    UpgradeTypes,
    ViewTypes,
};
