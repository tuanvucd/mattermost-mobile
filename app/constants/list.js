

const VISIBILITY_CONFIG_DEFAULTS = {
    viewAreaCoveragePercentThreshold: 3,
    waitForInteraction: false,
};

export default {
    VISIBILITY_CONFIG_DEFAULTS,
    VISIBILITY_SCROLL_DOWN: 'down',
    VISIBILITY_SCROLL_UP: 'up',
};
