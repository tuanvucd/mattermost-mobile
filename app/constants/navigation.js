// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import keyMirror from 'mattermost-redux/utils/key_mirror';

const NavigationTypes = keyMirror({
    NAVIGATION_RESET: null,
    NAVIGATION_CLOSE_MODAL: null,
    NAVIGATION_NO_TEAMS: null,
    RESTART_APP: null,
    NAVIGATION_ERROR_TEAMS: null,
    NAVIGATION_SHOW_OVERLAY: null,
});

export default NavigationTypes;
