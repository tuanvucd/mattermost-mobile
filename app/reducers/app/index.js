import {combineReducers} from 'redux';

import build from './build';
import version from './version';

export default combineReducers({
    build,
    version,
});
