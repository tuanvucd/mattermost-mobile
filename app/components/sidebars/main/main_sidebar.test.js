// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import {DeviceTypes} from 'app/constants';

import MainSidebar from './main_sidebar';

jest.mock('react-intl');

describe('MainSidebar', () => {
    const baseProps = {
        actions: {
            getTeams: jest.fn(),
            logChannelSwitch: jest.fn(),
            makeDirectChannel: jest.fn(),
            setChannelDisplayName: jest.fn(),
            setChannelLoading: jest.fn(),
        },
        blurPostTextBox: jest.fn(),
        currentTeamId: 'current-team-id',
        currentUserId: 'current-user-id',
        deviceWidth: 10,
        isLandscape: false,
        teamsCount: 2,
        theme: Preferences.THEMES.default,
    };

    test('should match, full snapshot', () => {
        const wrapper = shallow(
            <MainSidebar {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should not set the permanentSidebar state if not Tablet', () => {
        const wrapper = shallow(
            <MainSidebar {...baseProps}/>
        );

        wrapper.instance().handlePermanentSidebar();
        expect(wrapper.state('permanentSidebar')).toBeUndefined();
    });

    test('should set the permanentSidebar state if Tablet', async () => {
        const wrapper = shallow(
            <MainSidebar {...baseProps}/>
        );

        DeviceTypes.IS_TABLET = true;

        await wrapper.instance().handlePermanentSidebar();

        expect(wrapper.state('permanentSidebar')).toBeDefined();

        // Reset to false for subsequent tests
        DeviceTypes.IS_TABLET = false;
    });

    test('should re-render when the theme changes', () => {
        const theme = Preferences.THEMES.default;
        const newTheme = Preferences.THEMES.organization;
        const props = {
            ...baseProps,
            theme,
        };

        const wrapper = shallow(
            <MainSidebar {...props}/>
        );

        const instance = wrapper.instance();
        instance.render = jest.fn();

        expect(instance.render).toHaveBeenCalledTimes(0);
        wrapper.setProps({theme: newTheme});
        expect(instance.render).toHaveBeenCalledTimes(1);
    });
});
