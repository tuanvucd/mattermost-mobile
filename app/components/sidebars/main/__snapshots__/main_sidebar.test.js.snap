// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSidebar should match, full snapshot 1`] = `
<DrawerLayout
  drawerPosition="left"
  drawerWidth={-30}
  isTablet={false}
  onDrawerClose={[Function]}
  onDrawerOpen={[Function]}
  renderNavigationView={[Function]}
  useNativeAnimations={true}
/>
`;
